integration/tests/cook/util.py

```diff
@@ -681,15 +696,16 @@ def predicate(job):
         if not job['instances']:
             logger.info(f"Job {job_id} has no instances.")
         else:
-            inst = job['instances'][0]
-            if 'sandbox_directory' not in inst:
-                logger.info(f"Job {job_id} instance {inst['task_id']} has no sandbox directory.")
-            else:
-                logger.info(
-                    f"Job {job_id} instance {inst['task_id']} has sandbox directory {inst['sandbox_directory']}.")
-                return True
+            for inst in job['instances']:
+                if 'sandbox_directory' not in inst:
+                    logger.info(f"Job {job_id} instance {inst['task_id']} has no sandbox directory.")
+                else:
+                    logger.info(
+                        f"Job {job_id} instance {inst['task_id']} has sandbox directory {inst['sandbox_directory']}.")
+                    job['instance-with-sandbox'] = inst
```

Comment:
Should we handle `truthy` values returned from the predicate in `wait_until`? Then you could have just returned `inst` from this function.

Reply:
That seems like a pretty major change. A lot of our predicates are probably just returning `True`/`False`, yet the caller is expecting `wait_until` to return the result of the `query` function.

Reply:
Why would the helper function be better?

Reply:
We don't rely on the predicate function changing its input to extract the return value. We could just re-call the helper to extract the return value.

Reply:
> We don't rely on the predicate function changing its input to extract the return value

Is your concern a functional concern (it won't always work), or a stylistic concern?

Reply:
It is a stylistic concern. I also worry it sets a precedent for future usage patterns.

Reply:
I'm not really worried about this, and I think the current way is simpler, but if you feel strongly, I can change it.

Reply:
Done